store/db_schema.go

```diff
@@ -0,0 +1,160 @@
+package store
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common"
+)
+
+// dbSchemaRaw is the store model for a database schema.
+// Fields have exactly the same meaning as DBSchema.
+type dbSchemaRaw struct {
+	ID int
+
+	// Standard fields
+
+	// Related fields
+	DatabaseID int
+
+	// Domain specific fields
+	Metadata string
+	RawDump  string
+}
+
+func (raw *dbSchemaRaw) toDBSchema() *api.DBSchema {
+	return &api.DBSchema{
+		ID: raw.ID,
+
+		// Standard fields
+
+		// Related fields
+		DatabaseID: raw.DatabaseID,
+
+		// Domain specific fields
+		Metadata: raw.Metadata,
+		RawDump:  raw.RawDump,
+	}
+}
+
+// GetDBSchema gets the schema for a database.
+func (s *Store) GetDBSchema(ctx context.Context, databaseID int) (*api.DBSchema, error) {
+	cachedDBSchema := &api.DBSchema{}
+	ok, err := s.cache.FindCache(principalCacheNamespace, databaseID, cachedDBSchema)
+	if err != nil {
+		return nil, err
+	}
+	if ok {
+		return cachedDBSchema, nil
+	}
+
+	tx, err := s.db.BeginTx(ctx, nil)
+	if err != nil {
+		return nil, FormatError(err)
+	}
+	defer tx.Rollback()
+
+	// Build WHERE clause.
+	where, args := []string{"1 = 1"}, []interface{}{}
+	where, args = append(where, fmt.Sprintf("database_id = $%d", len(args)+1)), append(args, databaseID)
+
+	rows, err := tx.QueryContext(ctx, `
+		SELECT
+			id,
+			database_id,
+			metadata,
+			raw_dump
+		FROM db_schema
+		WHERE `+strings.Join(where, " AND "),
+		args...,
+	)
+	if err != nil {
+		return nil, FormatError(err)
+	}
+	defer rows.Close()
+
+	var list []*dbSchemaRaw
+	for rows.Next() {
+		var raw dbSchemaRaw
+		if err := rows.Scan(
+			&raw.ID,
+			&raw.DatabaseID,
+			&raw.Metadata,
+			&raw.RawDump,
+		); err != nil {
+			return nil, FormatError(err)
+		}
+		list = append(list, &raw)
+	}
+	if err := rows.Err(); err != nil {
+		return nil, FormatError(err)
+	}
+
```

Comment:
if err := tx.Commit()

Reply:
Good eye. Changed to avoid using txn at all.